functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -41,48 +41,36 @@
 @Validated
 public class ZeroMqSupplierProperties {
 
-	private static final SocketType DEFAULT_SOCKET_TYPE = SocketType.SUB;
-	private static final Duration DEFAULT_CONSUME_DELAY = Duration.ofSeconds(1);
-	private static final List<SocketType> VALID_SOCKET_TYPES =
-			Arrays.asList(SocketType.PAIR, SocketType.PULL, SocketType.SUB);
-
-	private final AtomicInteger bindPort = new AtomicInteger();
-
 	/**
 	 * The {@link SocketType} the connection should make.
-	 *
-	 * Allowable values:
-	 * - PAIR
-	 * - PULL
-	 * - SUB
 	 */
-	private SocketType socketType = DEFAULT_SOCKET_TYPE;
+	private SocketType socketType;
 
 	/**
-	 * Connection URL for {@link org.zeromq.ZMQ.Socket#connect(String)}.
-	 * Mutually exclusive with the {@link #bindPort}.
+	 * Connection URL for to the ZeroMQ TCP Server.
 	 */
 	private String connectUrl;
 
 	/**
-	 * A {@link Duration} to delay consumption when no data received.
+	 * Bind Port for connecting to the ZeroMQ TCP Server.
+	 */
+	private Integer bindPort;
```

Comment:
Can't it be just primitive `int` with the default meaning as `0`?
Of course, we should just ignore its value at all when `connectUrl` is provided.
Or you can also validate that port is not `> 0` when `connectUrl` is provided...